src/array_value.cpp

```diff
@@ -228,31 +228,69 @@ ValuePtr ArrayValue::map(Env *env, Block *block) {
     return new_array;
 }
 
-// TODO: accept integer and return array
-ValuePtr ArrayValue::first(Env *env) {
-    if (size() > 0) {
+ValuePtr ArrayValue::first(Env *env, ValuePtr n) {
+    auto has_count = n != nullptr;
+
+    if (!has_count) {
+        if (is_empty())
+            return NilValue::the();
+
         return (*this)[0];
-    } else {
-        return NilValue::the();
     }
+
+    ArrayValue* array = new ArrayValue();
+
+    n->assert_type(env, Value::Type::Integer, "Integer");
+    nat_int_t n_value = n->as_integer()->to_nat_int_t();
```

Comment:
When we have an argument which needs to be an int something is this the way to convert / check it?
I copied this code from `Array#shift` but was not sure about if this is correct enough.

Reply:
This is correct. This method only accepts an int (or nil), so if the user passes, say, a string, then they'll get the error:

```
TypeError (no implicit conversion of String into Integer)
```

If we have need to convert from String to Integer, we'd need to `send("to_i")`, but that's not what Ruby does here. So we're good! :smile: